### PR TITLE
Added serialization, no lookup yet

### PR DESCRIPTION
**Pull request by AdamMcCormick**
I added a hook that saves the game state as each puzzle is worked.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,15123 @@
...
+            }
+            return (
+                o(t, [
+                    {
+                        key: "connectedCallback",
+                        value: function () {
+                            var e,
+                                a = this;
+                            this.shadowRoot.appendChild(Sa.content.cloneNode(!0)),
+                                (this.shadowRoot.querySelector("#hash").textContent = null === (e = window.wordle) || void 0 === e ? void 0 : e.hash),
```
Comment (jonmcoe) [inline main.js:1019]: hash usage, bottom right display under Settings

File: main.js [hunk 2]
```diff
@@ -0,0 +1,15123 @@
...
+                        },
+                    ]),
+                    t
+                );
+            })(c(HTMLElement));
+        function Ca() {
+            dataLayer.push(arguments);
+        }
+        customElements.define("game-toast", Aa), (window.dataLayer = window.dataLayer || []), Ca("js", new Date());
+        Ca("config", "G-2SSGMHY3NP", { app_version: null === (Ea = window.wordle) || void 0 === Ea ? void 0 : Ea.hash, debug_mode: !1 });
```
Comment (jonmcoe) [inline main.js]: hash usage, google analytics

File: main.js [hunk 3]
```diff
@@ -0,0 +1,15123 @@
...
+            text: "".concat(l, "\n\n", d.trimEnd()),
+            hard: o,
+            status: e.gameStatus,
+        };
+    })(
+        JSON.parse(window.localStorage.getItem('gameState')),
+        JSON.parse(window.localStorage.getItem('darkTheme')),
+        JSON.parse(window.localStorage.getItem('colorBlindTheme'))
+    );
+    window.localStorage.setItem('saved'+share.day, JSON.stringify(share))
```
Comment (jonmcoe) [inline main.js:44]: is this your sole meaningful change to file? can you do a little more to point out what changes you made since formatting?

**Review by jonmcoe — CHANGES_REQUESTED** [review]
can you tell me more about the idea here? trying to store the share contents to localStorage's saved-XX right? any reason you prefer this over the entire gameState?

a few concerns
- on initial load, the localStorage is populated for a negative date. 
     - localStorage.getItem('saved-18797') '{"day":-18797,"text":"Wordle -18797 X/6\\n\\n","status":"IN_PROGRESS"}'
- on first solve, get the following error:
     - main.min.js:1 Uncaught TypeError: e.showStatsModal is not a function
    at main.min.js:1
- i dont need/want to properly version all this, but the hash specified in index.html does mean something (usages in line comments). if we're going to strip that, let's change it to `deadbeef` or something? could informally bump on any major change


curiosities
- how did you format to the js? really cool to see this more readable. that wont need to be done again, but i'd still be interested in knowing
- how is the minification done? that i WOULD want to do if i edit again. i dont think i've ever done on a single file like this, outside of a "proper" JS project with a package.json and whatnot

**Comment by jonmcoe** [comment]
uhhh oops. think you've addressed some of the above. let me repull and try it out again

**Comment by jonmcoe** [comment]
i'm indeed still getting the saved-18797 and the ` e.showStatsModal is not a function` preventing the stats from coming up after solving. on refresh, i get a working game n+1, with a saved0 that looks good and a saved1 in progress.

perhaps better to put a hook on game completion (either solving or 6th guess failure) that saves the just-solved problem's gameState to localStorage at that point? seems less risk of messing other things up, and more useful info (this would allow display of past games in app, and the share contents can always be regenerated from it)

**Comment by AdamMcCormick** [comment]
So the basic idea is that I replicated and modified the original game's "share" code into a global method that is then called whenever the state is updated. I had trouble locating a good place to hook for the end game but I can probably find one with a bit more inspection. To answer some specifics:
- I'll revise for initial game state. I can add the full state it just wasn't the use case I was going after.
- I'm not sure where the initial error is coming from but I can probably fix it
- I assumed the hash was generated from the code (a hash signature is pretty common) so it would change every release but be effectively random
- There are lots of JS "beautifier" tools that de-uglify JS and "uglifiers" that do minimization. I used https://beautifier.io/ and https://www.toptal.com/developers/javascript-minifier/

**Comment by jonmcoe** [comment]
i suspect the hash **was** generated from code contents, but i have not bumped it for any of my updates. this isnt super important, we'd be fine leaving it as c1506a22 forever. just figured since you removed that c1506a22 from filename we may as well set it (arbitrarily) to be something different. certainly wouldn't hold up a merge over this.

i think i'm still missing something about your objective here. why do we want to save anything additional, at any time, besides completion of game? i had thought the goal was to be able to revisit past games for in app viewing, to generate the sharing copyable, or both. is there something additional? i think saving a savedX with gameState at time of completion would be sufficient for that (tho obviously UI questions remain)

File: main.js (hunk 3 above)
Comment (AdamMcCormick) [reply main.js:44]: Adding this method and calling it down below are the only intentional changes.

File: main.js (hunk 1 above)
Comment (AdamMcCormick) [reply main.js:1019]: I changed the hash to `deadbeef` but I can revert if you'd prefer

File: main.js (hunk 2 above)
Comment (AdamMcCormick) [reply main.js]: Yeah, might make sense to strip the GA code entirely unless you are using your own

**Comment by AdamMcCormick** [comment]
Mostly because it removes the need for additional UI to get the share string. The day was for human readability because it's generated from the date as part of sharing. My last change adds the full state to the object so we could set the whole game state if desired.

**Comment by jonmcoe** [comment]
ok sure, it's not big. might as well keep share stored as well if you prefer. can you tell me more about why you'd update these on every state change rather than solely at completion of game?

**Comment by jonmcoe** [comment]
ok sure, it's not big. might as well keep share stored as well if you prefer. can you tell me more about why you'd update these on every state change rather than solely at completion of game?

**Comment by jonmcoe** [comment]
ok sure, it's not big. might as well keep share stored as well if you prefer. can you tell me more about why you'd update these on every state change rather than solely at completion of game?

File: main.js (hunk 2 above)
Comment (jonmcoe) [reply main.js]: i was able to open a new project and get the existing GA events for nearly-free.
Screen Shot 2022-01-20 at 5 26 33 PM

**Comment by AdamMcCormick** [comment]
It was just an easier place to tie in to the code because of the way the original minification changed the win condition structure.

File: main.js [hunk 4]
```diff
@@ -0,0 +1,15125 @@
...
+                "meats",
+                "mebos",
+                "mechs",
+                "mecks",
+                "medii",
+                "medle",
+                "meeds",
+                "meers",
+                "meets",
+                "meffs",
```
Comment (jonmcoe) [inline main.js:14350]: these are the toasts upon game completion. GENIUS, PHEW, etc on win, solution word on loss. this is probably good place to set our new localStorage elements if you decvide to do it on close of game only

Comment (AdamMcCormick) [reply main.js:14350]: I hooked the initial stats display which seems to work just fine. It won't run on restore from local storage but the result should be saved at that point

**Pull request closed without merging**